File: flutter/android/src/main/kotlin/com/veryoauthsdk/flutter/VeryOauthSDKPlugin.kt
package com.veryoauthsdk.flutter

import android.content.Context
import com.veryoauthsdk.*
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class VeryOauthSDKPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
    private lateinit var channel: MethodChannel
    private var context: Context? = null
    private var activity: android.app.Activity? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "very_oauth_sdk")
        channel.setMethodCallHandler(this)
        context = flutterPluginBinding.applicationContext
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivity() {
        activity = null
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "authenticate" -> {
                authenticate(call, result)
            }
            "cancelAuthentication" -> {
                cancelAuthentication(result)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    private fun authenticate(call: MethodCall, result: Result) {
        val args = call.arguments as? Map<String, Any> ?: return result.error("INVALID_ARGUMENTS", "Invalid arguments", null)
        
        val clientId = args["clientId"] as? String ?: return result.error("INVALID_ARGUMENTS", "clientId is required", null)
        val redirectUri = args["redirectUri"] as? String ?: return result.error("INVALID_ARGUMENTS", "redirectUri is required", null)
        val userId = args["userId"] as? String
        val authorizationUrl = args["authorizationUrl"] as? String
        val scope = args["scope"] as? String
        val browserModeIndex = args["browserMode"] as? Int ?: 1 // Default to webview
        val language = args["language"] as? String

        val browserMode = when (browserModeIndex) {
            0 -> BrowserMode.SYSTEM_BROWSER
            1 -> BrowserMode.WEBVIEW
            else -> BrowserMode.WEBVIEW
        }

        val config = OAuthConfig(
            clientId = clientId,
            redirectUri = redirectUri,
            userId = userId,
            authorizationUrl = authorizationUrl ?: "https://connect.very.org/oauth/authorize",
            scope = scope,
            browserMode = browserMode,
            language = language
        )

        val activity = this.activity ?: return result.error("NO_ACTIVITY", "No activity available", null)

        VeryOauthSDK.getInstance().authenticate(
            context = activity,
            config = config,
            callback = { oauthResult ->
                val resultMap = mapOf(
                    "code" to oauthResult.code,
                    "error" to oauthResult.error.ordinal
                )
                result.success(resultMap)
            }
        )
    }

    private fun cancelAuthentication(result: Result) {
        VeryOauthSDK.getInstance().cancelAuthentication()
        result.success(null)
    }
}

File: flutter/ios/Classes/VeryOauthSDKPlugin.swift
import Flutter
import UIKit
import VeryOauthSDK

public class VeryOauthSDKPlugin: NSObject, FlutterPlugin {
    private var currentViewController: UIViewController?
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "very_oauth_sdk", binaryMessenger: registrar.messenger())
        let instance = VeryOauthSDKPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "authenticate":
            authenticate(call: call, result: result)
        case "cancelAuthentication":
            cancelAuthentication(result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
    private func authenticate(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any] else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "Invalid arguments", details: nil))
            return
        }
        
        guard let clientId = args["clientId"] as? String else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "clientId is required", details: nil))
            return
        }
        
        guard let redirectUri = args["redirectUri"] as? String else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "redirectUri is required", details: nil))
            return
        }
        
        let userId = args["userId"] as? String
        let authorizationUrl = args["authorizationUrl"] as? String
        let scope = args["scope"] as? String
        let browserModeIndex = args["browserMode"] as? Int ?? 1 // Default to webview
        let language = args["language"] as? String
        
        let browserMode: BrowserMode = browserModeIndex == 0 ? .systemBrowser : .webview
        
        let config = OAuthConfig(
            clientId: clientId,
            redirectUri: redirectUri,
            authorizationUrl: authorizationUrl,
            scope: scope,
            browserMode: browserMode,
            userId: userId,
            language: language
        )
        
        guard let rootViewController = UIApplication.shared.windows.first?.rootViewController else {
            result(FlutterError(code: "NO_VIEW_CONTROLLER", message: "No root view controller available", details: nil))
            return
        }
        
        currentViewController = rootViewController
        
        VeryOauthSDK.shared.authenticate(
            config: config,
            presentingViewController: rootViewController
        ) { [weak self] oauthResult in
            DispatchQueue.main.async {
                let resultMap: [String: Any] = [
                    "code": oauthResult.code,
                    "error": oauthResult.error.rawValue
                ]
                result(resultMap)
                self?.currentViewController = nil
            }
        }
    }
    
    private func cancelAuthentication(result: @escaping FlutterResult) {
        VeryOauthSDK.shared.cancelAuthentication()
        result(nil)
    }
}
